plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/Layered.melk

```diff
@@ -345,14 +346,36 @@ group crossingMinimization {
 
         advanced option type: GreedySwitchType {
             label "Greedy Switch Crossing Minimization"
-            description
+            description 
                 "Greedy Switch strategy for crossing minimization. The greedy switch heuristic is executed 
-                 after the regular layer sweep as a post-processor."
+                 after the regular crossing minimization as a post-processor. 
+                 Note that if 'hierarchyHandling' is set to 'INCLUDE_CHILDREN', the 'greedySwitchHierarchical.type' 
+                 option must be used."
             default = GreedySwitchType.TWO_SIDED
             targets parents
+            requires crossingMinimization.strategy == CrossingMinimizationStrategy.LAYER_SWEEP
+            requires org.eclipse.elk.hierarchyHandling == HierarchyHandling.SEPARATE_CHILDREN
```

Comment:
Will this work for graphs that don't have their hierarchy handling set? Is this the default option?

Reply:
In fact, the default is `INHERIT` (which is turned into `SEPARATE_CHILDREN` during `RecursiveGraphLayoutEngine#evaluateHierarchyHandlingInheritance(...)`). 

As a validation of the requires (although I'm not sure if we've something like it implemented right now) would probably happen beforehand, the line would be problematic. 

I guess I'll remove the line. Unfortunately there's no `!=` - which is the task of another long-standing ticket.

Reply:
Sounds good.